### PR TITLE
Adjust deployment pkg return consistency

### PR DESCRIPTION
**Pull request by sebrandon1**
Broken out from: https://github.com/openshift-kni/eco-goinfra/pull/355

### CODE REVIEW

File: pkg/deployment/deployment.go [hunk 1]
```diff
@@ -69,18 +69,24 @@ func NewBuilder(
 		glog.V(100).Infof("The name of the deployment is empty")
 
 		builder.errorMsg = "deployment 'name' cannot be empty"
+
+		return &builder
```
Comment (kononovn) [inline pkg/deployment/deployment.go:73]: No need for this repeated return. We have validate method which will print that builder has error and return the exact error.

File: pkg/deployment/deployment.go [hunk 2]
```diff
@@ -451,7 +457,7 @@ func (builder *Builder) CreateAndWaitUntilReady(timeout time.Duration) (*Builder
 	if _, err := builder.Create(); err != nil {
 		glog.V(100).Infof("Failed to create deployment. Error is: '%s'", err.Error())
 
-		return nil, fmt.Errorf(err.Error())
+		return nil, err
```
Comment (kononovn) [inline pkg/deployment/deployment.go:460]: +1

**Review by kononovn — CHANGES_REQUESTED** [review]
1 minor comment

File: pkg/deployment/deployment.go (hunk 1 above)
Comment (kononovn) [reply pkg/deployment/deployment.go:73]: We use builder.errorMsg in our unit test. You just need to check that builder has relevant error.

Comment (sebrandon1) [reply pkg/deployment/deployment.go:73]: I think we've talked about this before for other parts of the code where you could potentially have a blank string for name, namespace and an empty `if len(labels) == 0 {` and the last condition would be the builder.errorMsg that was set.

Comment (kononovn) [reply pkg/deployment/deployment.go:73]: yes and It's OK because code will not work until all errors I'll be fixed.

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**